src/Game.ts

```
@@ -1504,7 +1504,7 @@ export class Game implements ISerializable<SerializedGame> {
         return player;
       }
     }
-    throw new Error('No player has played requested card');
+    return undefined;
```

I think I understood this yesterday but why does this have to change for `Habitat Martae`? Throwing here is helpful for failing fast when the player is not found. Most places we call this we expect to get a result. Is there a new place we are calling in this change where we don't expect to get a return value?

---

Look at src/cards/pathfinders/CollegiumCopernicus.ts. Failing is now considered a valid workable case and not just a system error. I would need to change that behavior to try/catch. And that's fine, but it's not the pattern we use in this app.

---

Can you link to where in `CollegiumCopernicus` this `getCardPlayer` is called? I don't see a reference to `getCardPlayer` in that file.

---

Sorry, you're right. I don't see a specific place where this was necessary.

Thinking back, I bet I was doing because `owner?.` is pretty small, and just as safe. Perhaps it's hiding an error, but, well, I don't know.

---

If you think restoring the original behavior is better, I will. But knowing so much about when this is called and why, this is a pretty safe change.

---

> > Hang back on approval. I want to write some more thoughts.
> 
> Is this ready for approval?

It is now.